wgengine/netstack/netstack.go

```diff
@@ -189,14 +189,18 @@ func (ns *Impl) updateDNS(nm *netmap.NetworkMap) {
 func (ns *Impl) addSubnetAddress(pn tcpip.NetworkProtocolNumber, ip netaddr.IP) {
 	ns.mu.Lock()
 	ns.connsOpenBySubnetIP[ip]++
+	// Only register address into netstack for first concurrent connection.
+	if ns.connsOpenBySubnetIP[ip] == 1 {
+		ns.ipstack.AddAddress(nicID, pn, tcpip.Address(ip.IPAddr().IP))
```

Comment:
probably doesn't matter but this doesn't really need to be done while ns.mu is held. 

```
needAdd := ns.connsOpenBySubnetIP[ip] == 1
ns.mu.Unlock
if needAdd {
     ns.ipstack.AddAddress(nicID, pn, tcpip.Address(ip.IPAddr().IP))
}
```